hearth/lib/hearth/telemetry/base.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Telemetry
+    # Base for TelemetryProvider classes.
```

Comment:
I'd add a few more lines to these descriptions. What does the provider do? What does the tracer provider do?